packages/plugins/geth/src/check.js

```diff
@@ -5,6 +5,9 @@ const LIVENESS_CHECK=`{"jsonrpc":"2.0","method":"web3_clientVersion","params":[]
 
 const parseAndRespond = (data, cb) => {
   const resp = JSON.parse(data);
+  if (!resp || !resp.result) {
```

Comment:
Is it guaranteed that `data` will be parse-able by `JSON.parse`? If not, should we wrap it in try/catch?

Reply:
Agreed, good point

Reply:
Updated. Also added the same fixes for parity